internal/interpreter/value_model.rs

```diff
@@ -115,3 +115,32 @@ impl BoolModel {
         }
     }
 }
+
+// A map model that wraps a Model
+pub struct ValueMapModel<T>(pub ModelRc<T>);
+
+impl<T: TryFrom<Value> + Into<Value> + 'static> Model for ValueMapModel<T> {
+    type Data = Value;
+
+    fn row_count(&self) -> usize {
+        self.0.row_count()
+    }
+
+    fn row_data(&self, row: usize) -> Option<Self::Data> {
+        self.0.row_data(row).map(|x| x.into())
+    }
+
+    fn model_tracker(&self) -> &dyn ModelTracker {
+        self.0.model_tracker()
+    }
+
+    fn as_any(&self) -> &dyn core::any::Any {
+        self
```

Comment:
Shouldn't that forward to the inner model?

Reply:
We do the downcast here:
https://github.com/slint-ui/slint/pull/8575/files#diff-3aedfd36abe5a45b6ca0eec3be66bfcf4e6e94db26b7ea9e93218a6a03a4082dR445